ocp_resources/resource.py

```diff
@@ -1017,6 +1030,26 @@ def to_yaml(self):
         self.logger.info(f"\n{resource_yaml}")
         return resource_yaml
 
+    def hash_resource_dict(self, resource_dict: dict):
+        if hasattr(self, "keys_to_hash"):
+            return self.hash_resource_dict_rec(nested_dict=copy.deepcopy(resource_dict))
+        return resource_dict
+
+    def hash_resource_dict_rec(self, nested_dict: dict):
+        if not isinstance(nested_dict, dict):
+            return nested_dict
+
+        for key in nested_dict.keys():
+            if key in self.keys_to_hash:
+                if isinstance(nested_dict[key], dict):
+                    nested_dict[key] = {_key: "***" for _key in nested_dict[key].keys()}
+                else:
+                    nested_dict[key] = "***"
+            else:
+                nested_dict[key] = self.hash_resource_dict_rec(nested_dict[key])
+
```

Comment:
no need for this; keys_to_hash should be a list of absolute paths, e.g ["spec.data", "metadata.name"]
Then use python-benedict to get the value from the dict and hash it; no need for recursive function.

Reply:
benedict does not support dictionary keys that contain `"."` character:
```
ValueError: Key should not contain keypath separator '.', found: '.dockerconfigjson'.
```

Reply:
done